File: selection_sort.js
my_list = [6, 50, 22, 8, 30, 1, 4, 7, 9];


function selection_sort(arr){

  for(var i = 0; i< arr.length; i++){
    var val_index = i; //store index of i
    var min = arr[i];  //store value of min
    for(var j =i; j<arr.length; j++){
     //use two loops. 1 to hold variable and the other to loop through arr for comparison
     //find min, store index 
      if(arr[j] <= min){
        min = arr[j];
        min_index = j;
        }    
        if(j == arr.length-1){
          after_swap = swap(arr, val_index, min_index);
          console.log(after_swap);
          }
      }
    }
}

function swap(arr, val_index, min_index){
  temp = arr[val_index];
  arr[val_index] = arr[min_index];
  arr[min_index] = temp;
  return arr;
}


selection_sort(my_list);

File: selection_sort.py
#selection sort

#start with linear selection
#iterate through array and find minimum value
#swap variable position with minimum value

my_list = [6, 5, 2, 8, 3, 1, 4, 7, 9]
print "length of list %s" % len(my_list)
def selection_sort(my_list):
	leng = len(my_list)
	for i in range(leng):
		min = my_list[i]
		val_index = i
		#print "i %s" % i
		for j in range(i,leng):
			#print "j %s" % j
			if my_list[j] <= min:
				min = my_list[j]
				min_index = j
			if j == leng-1:
				#value of index is = i
				#min_index = index of the found min in my list
				after_swap = swap(my_list, val_index, min_index)
	print after_swap



def swap(my_list, val_index, min_index):

	temp = my_list[val_index]
	my_list[val_index] = my_list[min_index]
	my_list[min_index] = temp
	return my_list



selection_sort(my_list)